### PR TITLE
Use stems in Insights

### PR DESCRIPTION
**Pull request by nataliashitova**
## Summary

This PR can be summarized in the following changelog entry:

* Changes the way Insights are generated: Instead of using word combinations that occurred in the text, splits the text in words, normalizes them by stem and only then determines which words should be displayed in the Insights.

## Relevant technical choices:

* Adds safety checks to the functions that process `prominent words`, so that they do not fail in edge cases.

## Test instructions

Please see the other PR in the issue for testing instructions.

Fixes https://github.com/Yoast/wordpress-seo-premium/issues/2292

### CODE REVIEW

File: packages/yoast-components/composites/KeywordSuggestions/KeywordSuggestions.js [hunk 1]
```diff
@@ -110,7 +110,7 @@ KeywordSuggestions.propTypes = {
 };
 
 KeywordSuggestions.defaultProps = {
-	keywordLimit: 5,
+	keywordLimit: 20,
```
Comment (manuelaugustin) [inline packages/yoast-components/composites/KeywordSuggestions/KeywordSuggestions.js]:
```suggestion
	keywordLimit: 5,
```

Comment (nataliashitova) [reply packages/yoast-components/composites/KeywordSuggestions/KeywordSuggestions.js]: 👍

File: packages/yoastseo/src/stringProcessing/relevantWords.js [hunk 2]
```diff
@@ -26,6 +26,20 @@ function getRelevantCombinations( wordCombinations ) {
...
 }
 
+/**
+ * Returns only those relevant combinations that occur 5 or more times and do not consist of special characters.
+ *
+ * @param {WordCombination[]} wordCombinations A list of word combinations.
+ *
+ * @returns {WordCombination[]} Only relevant word combinations.
+ */
+function getRelevantCombinationsForInsights( wordCombinations ) {
```
[the remark this answers is not in the document]
Comment (hansjovis) [reply packages/yoastseo/src/stringProcessing/relevantWords.js]: This would also solve the possible bug that a `WordCombination` could get on the list of relevant words when it has 0 or less occurrences 😉.

File: packages/yoastseo/src/stringProcessing/relevantWords.js [hunk 3]
```diff
@@ -224,6 +246,7 @@ export default {
 	getRelevantWords: getRelevantWords,
 	getRelevantWordsFromPaperAttributes: getRelevantWordsFromPaperAttributes,
 	getRelevantCombinations: getRelevantCombinations,
+	getRelevantCombinationsForInsights: getRelevantCombinationsForInsights,
```
Comment (hansjovis) [inline packages/yoastseo/src/stringProcessing/relevantWords.js]: I think this can be simplified as thus:
```suggestion
	getRelevantCombinationsForInsights,
```

File: packages/yoastseo/src/stringProcessing/relevantWords.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (nataliashitova) [reply packages/yoastseo/src/stringProcessing/relevantWords.js]: Fair point! Fixed it.

**Comment by hansjovis** [comment]
Acceptance test: ✅ 

(See https://github.com/Yoast/wordpress-seo-premium/pull/2296)
